src/ui/router/ApplicationRouter.js

```diff
@@ -57,11 +41,90 @@ class ApplicationRouter extends EventEmitter {
      * route(path, handler);
      * start(); Start routing.
      */
-    constructor() {
+    constructor(openmct) {
         super();
+
+        this.locationBar = new LocationBar();
+        this.openmct = openmct;
         this.routes = [];
         this.started = false;
-        this.locationBar = new LocationBar();
+
+        this.setHash = _.debounce(this.setHash.bind(this), 300);
+    }
+
+    // Public Methods
+
+    destroy() {
+        this.locationBar.stop();
+    }
+
+    deleteSearchParam(paramName) {
```

Comment:
These functions need some documentation. It's not clear to me what they do, or what the difference is between them all.

Reply:
added JSDocs